Switch to certbot binary

morrismukiri opened the pull request:
Closes #3

morrismukiri committed:
Switch to certbot binary

morrismukiri commented:
Required for https://github.com/onaio/playbooks/issues/1047

jasonrogena commented on tasks/main.yml:
```diff
@@ -1,39 +1,50 @@
+    job: /usr/bin/certbot renew --prehook "service {{ certbot_webserver_installed }} stop" --post-hook "service {{ certbot_webserver_installed }} start"
+  when:
+    - certbot_create_certs
+    - certbot_plugin == "standalone"
```
@morrismukiri does this mean that autorenewal will only happen if you use certbot in standalone mode?

jasonrogena replied:
After autorenewal, will it be up to the NGINX role, for instance, to copy over the new files?

morrismukiri replied:
certbot automatically creates a systemd timer that runs twice daily so we don't need to create a cron job. The renew command uses  the same strategy used to generate the certificate. The reason standalone is special is because certbot won't know to stop the web server(as we do when creating the cert), so we create a cron job that is going to stop the web server and start it once renewal is done.

morrismukiri replied:
The role assumes that if you don't want certbot to install the cert into the site configurations, you already have that figured out; either by copying the certs from the letsencrypt directory or using the path directly in the site configuration. Otherwise certbot will modify the site configuration to add the certs and redirect.

jasonrogena replied:
Cool. Makes sense.

morrismukiri committed:
Allow getting certificates for multiple domains
Make certificate generation more configurable

jasonrogena committed:
Add Support For Systemd Timers

Add support for renewing certificates using systemd timers.

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
Fix Template Issue & Default Systemd Variable

Fix issue causing role to fail because of how the systemd files were
being copied.

Set the default value for certbot_renew_certs_systemd_oncalendar to
something that is actually valid.

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

morrismukiri committed:
Fix trailing comma on domain list

morrismukiri committed:
Merge branch 'use-certbot-binary' of github.com:onaio/ansible-certbot into use-certbot-binary

jasonrogena committed:
Make Sure systemd Timer Is Restarted

Make sure the systemd time is restarted when installed.

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>
